packages/hardhat-core/src/internal/solidity/compiler/downloader.ts

```diff
@@ -8,7 +8,7 @@ import { promisify } from "util";
 import { download } from "../../util/download";
 import { assertHardhatInvariant, HardhatError } from "../../core/errors";
 import { ERRORS } from "../../core/errors-list";
-import { Mutex } from "../../vendor/await-semaphore";
+import { MultiProcessMutex } from "../../vendor/await-semaphore";
```

Comment:
We should rename this file since it's no longer a vendored implementation.

Reply:
Ah, we are not removing the vendored code. In that case, we should create a separate file for the multi-process mutex.

Reply:
I moved the code for the multi-process-mutex in a different file